### PR TITLE
Add TranscodingProfile conditions

### PR DESCRIPTION
**Pull request by dmitrylyzo**
The idea is to add some control over the transcoding profile separately from the direct profiles.
For example, no resolution limit for DirectPlay, but no more screen size for transcoding.
Ideally, we need global conditions as well.

**Changes**
Add TranscodingProfile conditions

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

**Comment by Shadowghost** [comment]
Does this also work as intended with #7325 instead of #6454?

**Comment by dmitrylyzo** [comment]
> Does this also work as intended with #7325 instead of #6454?

I haven't tested with them at all.
Now I tried with #7325 - it seems to work (but I need to move the call).

**Review by crobibero — APPROVED** [review]

**Comment by dmitrylyzo** [comment]
Hmm, this breaks remuxing (it is also `Transcode`) :thinking:

File: MediaBrowser.Model/Dlna/StreamBuilder.cs [hunk 1]
```diff
@@ -710,6 +710,11 @@ private StreamInfo BuildVideoItem(MediaSourceInfo item, VideoOptions options)
                     if (playlistItem.PlayMethod != PlayMethod.DirectPlay)
                     {
                         playlistItem.PlayMethod = PlayMethod.Transcode;
+
+                        if ((playlistItem.TranscodeReasons & (VideoReasons | TranscodeReason.ContainerBitrateExceedsLimit)) != 0)
```
Comment (dmitrylyzo) [inline MediaBrowser.Model/Dlna/StreamBuilder.cs:714]: Or maybe `VideoReasons' is enough (since that means using another container)?

**Review by joshuaboniface — APPROVED** [review]

**Pull request merged**